concrete/src/Service/Detector/HTTP/ApacheDetector.php

```diff
@@ -29,19 +29,37 @@ public function __construct(Request $request)
     public function detect()
     {
         $result = null;
-        if ($result === null && $this->request->server->has('SERVER_SOFTWARE')) {
-            $result = $this->detectFromServer($this->request->server->get('SERVER_SOFTWARE'));
+        if (($result === null || $result === '') && $this->request->server->has('SERVER_SOFTWARE')) {
+            $version = $this->detectFromServer($this->request->server->get('SERVER_SOFTWARE'));
+            if ($version !== null) {
+                $result = $version;
+            }
         }
-        if ($result === null && function_exists('apache_get_version')) {
-            $result = $this->detectFromSPL(@apache_get_version());
+        if (($result === null || $result === '') && function_exists('apache_get_version')) {
+            $version = $this->detectFromSPL(@apache_get_version());
+            if ($version !== null) {
+                $result = $version;
+            }
         }
-        if ($result === null) {
+        if (($result === null || $result === '') && defined('apache_get_version')) {
```

Comment:
`$result === null` means *Apache not (yet) detected*
`$result === ''` means *Apache detected, but version unknown*

So,
```
if (($result === null || $result === '') && ...) {
```
means: let's try to (better) detect the apache version